### PR TITLE
swift-format 0.50700.1

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
The `inreplace` needs to be removed (or fixed).

**Comment by Bo98** [comment]
I suspect this will need Swift 5.7.1 which is not released yet. It will be when Xcode 14.1 is out this month.

**Comment by lovetodream** [comment]
> I suspect this will need Swift 5.7.1 which is not released yet. It will be when Xcode 14.1 is out this month.
> 
> This looks like it needs the new `lib_InternalSwiftSyntaxParser` from 5.7.1.

It does still work with Swift 5.7 as shown in the release notes https://github.com/apple/swift-format/releases/tag/0.50700.1

**Comment by chkpnt** [comment]
Xcode 14.1 is now available. :-)

**Comment by Bo98** [comment]
> > I suspect this will need Swift 5.7.1 which is not released yet. It will be when Xcode 14.1 is out this month.
> > This looks like it needs the new `lib_InternalSwiftSyntaxParser` from 5.7.1.
> 
> It does still work with Swift 5.7 as shown in the release notes https://github.com/apple/swift-format/releases/tag/0.50700.1

As a language sure, but it requires a new lib_InternalSwiftSyntaxParser which ships with Xcode.

Thankfully, I think this is going away with Swift 5.8 hopefully.

Anyhow bundling this in with #114549.

**Comment by Bo98** [comment]
Or perhaps it didn't make it to Swift 14.1:

```
<unknown>: error: Unable to format test.swift: The loaded '_InternalSwiftSyntaxParser' library is from a toolchain that is not compatible with this version of SwiftSyntax
```

**Comment by chkpnt** [comment]
After removing the line inreplace-line as stated in the comment, I was able to build 0.50700.1 without any problems:

```
 ich@asiago-lbs  /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula   master ±  swift --version
swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: x86_64-apple-macosx12.0
 ich@asiago-lbs  /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula   master ±  xcode-select -p
/Applications/Xcode_14.1.app/Contents/Developer
 ich@asiago-lbs  /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula   master ±  brew reinstall --build-from-source swift-format
==> Cloning https://github.com/apple/swift-format.git
Updating /Users/ich/Library/Caches/Homebrew/swift-format--git
==> Checking out tag 0.50700.1
HEAD is now at 5f18422 Merge pull request #413 from allevato/point-one
HEAD is now at 5f18422 Merge pull request #413 from allevato/point-one
==> Reinstalling swift-format
==> swift build --disable-sandbox -c release -Xlinker -rpath -Xlinker /Library/Developer/CommandLineTools/usr/lib/swift/macosx
🍺  /usr/local/Cellar/swift-format/0.50700.1: 6 files, 15.9MB, built in 3 minutes 42 seconds
==> Running `brew cleanup swift-format`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /Users/ich/Library/Caches/Homebrew/swift-format--0.50700.0... (3.1MB)
```

File: Formula/swift-format.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class SwiftFormat < Formula
   desc "Formatting technology for Swift source code"
   homepage "https://github.com/apple/swift-format"
   url "https://github.com/apple/swift-format.git",
-      tag:      "0.50700.0",
-      revision: "3dd9b517b9e9846435aa782d769ef5825e7c2d65"
+      tag:      "0.50700.1",
+      revision: "5f184220d032a019a63df457cdea4b9c8241e911"
   license "Apache-2.0"
   version_scheme 1
   head "https://github.com/apple/swift-format.git", branch: "main"
```
Comment (Gaziav) [inline Formula/swift-format.rb:9]: can you please change  
**depends_on xcode: ["14.0", :build**] 
to  
**depends_on xcode: ["14.1", :build]**

Comment (Porkepix) [reply Formula/swift-format.rb:9]: Done. However I believe maintainers wanted to update it along the last swift update.

Comment (Bo98) [reply Formula/swift-format.rb:9]: > However I believe maintainers wanted to update it along the last swift update.

As above, turns out Xcode 14.1 isn't actually compatible with this release. So decided to go for a partial update for Linux only and keep this open to track an eventual macOS update.

**Comment by barksten** [comment]
Any updates on this? I was about to create a new PR before I found this

**Comment by Bo98** [comment]
swift-format is tightly coupled with `lib_InternalSwiftSyntaxParser.dylib` (found at `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/lib_InternalSwiftSyntaxParser.dylib`).

0.50700.0 is compatible with the dylib shipped with Xcode 14.0 and 14.1.
0.50700.1 is compatible with neither.

I've not checked Xcode 14.2 so maybe they've updated it finally.

The only change to 0.50700.1, the `if let` fix, required changes to `lib_InternalSwiftSyntaxParser.dylib`.

**Comment by Arlaz** [comment]
Doesn't work with Xcode 14.2 either
Will probably have to wait to the next version of swift-format, given the following note :

> NOTE: `swift-format` on the `main` branch now uses a version of
> [SwiftSyntax](https://github.com/apple/swift-syntax) whose parser has been
> rewritten in Swift and no longer has dependencies on libraries in the
> Swift toolchain. This allows `swift-format` to be built, developed, and
> run using any version of Swift that can compile it, decoupling it from
> the version that supported a particular syntax.

**Comment by Bo98** [comment]
Yeah, we will always align with what Xcode ships since that's the Swift distribution most developers will be using. If you are using a swift.org toolchain, then sorry there's not much we can do as a version that is compatible with both does not exist since 5.7.1.

Next version of Xcode will likely be when Swift 5.8 launches in March and we can finally put this strong coupling of Xcode versions to swift-format versions behind us.

**Pull request closed without merging**